### ISSUE TITLE
Getting OOM (out of memory) when using big files

### ISSUE THREAD

**Issue by shamamayair** [SYSTRAN/faster-whisper#1206]
When using big files we getting out of memory did some debug and found it happen here:

https://github.com/SYSTRAN/faster-whisper/blob/1b24f284c9b39bcdf47233d7ae507abaea7a98c8/faster_whisper/feature_extractor.py#L189

its only happen on big files (few hours) on small files its working fine.

this is very basic draft of the code we using (we fo course getting the oom on the transcribe:

model = WhisperModel(model_size)

vo = VadOptions(
    min_speech_duration_ms=250,
    speech_pad_ms=30,
    min_silence_duration_ms=5000,
)

options = {
    "word_timestamps": True,
    "vad_filter": True,
    "condition_on_previous_text": False,
    "hallucination_silence_threshold": 3,
    "log_prob_threshold": -0.5,
}

options["vad_parameters"] = vo

segments, info = model.transcribe(audio_file, **options)

did you encounter such an issue ? any ideas?

**Comment by Purfview** [comment]
It was fixed in the last commit: https://github.com/SYSTRAN/faster-whisper/pull/1198

Install the latest master:
Press "Code" button then press "Downlod ZIP", then run:  `pip install "faster-whisper-master.zip`
Or with git: `pip install git+https://github.com/SYSTRAN/faster-whisper.git`

**Comment by MahmoudAshraf97** [comment]
> It was fixed in the last commit: #1198
> 
> Install the latest master: Press "Code" button then press "Downlod ZIP", then run: `pip install "faster-whisper-master.zip"` Or with git: `pip install git+https://github.com/SYSTRAN/faster-whisper.git`

I guess this might be more related to the feature extraction rather than VAD, but if the vad used less memory that might help with the problem but it will still be there

**Comment by jhj0517** [comment]
Can you please reopen this? According to https://github.com/jhj0517/Whisper-WebUI/issues/424#issuecomment-2557960708, this seems to still be reproducible on Colab with the large file ( 1 hour 40 minutes).

**Comment by Purfview** [comment]
> Can you please reopen this? According to [jhj0517/Whisper-WebUI#424 (comment)](https://github.com/jhj0517/Whisper-WebUI/issues/424#issuecomment-2557960708), this seems to still be reproducible on Colab with the large file ( 1 hour 40 minutes).

Maybe it was on low memory even before VAD, what RAM usage it shows without VAD?

**Comment by jhj0517** [comment]
@Purfview I just tried to reproduce it myself on my side with 2 hours of video, but the CPU RAM was reached only 5.4 GB.
So I guess the problem is not related to this issue, it's a problem on my side. I'm sorry for giving you confusion.

This is the peak CPU RAM data I observed with 2 hours of audio (the left "시스템 RAM" means CPU RAM):

1) when using VAD together for 2 hours of audio : 5.4GB peak RAM (CPU)
![When using VAD for 2hours of audio](https://github.com/user-attachments/assets/67f64019-2a03-4958-bc3a-5d146a8bb2c4)

2) when using only faster-whisper for 2 hours of audio: 3.5GB peak RAM (CPU)
![When using faster-whisper only for 2hours of audio](https://github.com/user-attachments/assets/7d123ebf-c955-4e5e-a0bc-f95937167511)
